dataactvalidator/filestreaming/csvReader.py

```diff
@@ -67,12 +83,24 @@ def get_writer(bucket_name, filename, header, is_local, region=None):
         """
         Gets the write type based on if its a local install or not.
         """
+        # TODO verify that the writer works corectly for local and S3
         if is_local:
             return CsvLocalWriter(filename, header)
         if region is None:
             region = CONFIG_BROKER["aws_region"]
         return CsvS3Writer(region, bucket_name, filename, header)
 
+    def _transfer_s3_file_to_local(self, bucket, filename):
+        (file, file_path) = tempfile.mkstemp()
+        file.close()
+        os.close(file_path)
+        os.remove(file_path)
```

Comment:
If we're closing and removing the temp file before writing to it, would it still be used? I would think it would just create a new file with `file_path` when calling `open`. Wouldn't we want to delete this after we're done reading from it for the validations, etc.?